tests/potential/test_potential.py

```diff
@@ -581,5 +581,211 @@ def test_saveload(self):
+        np.testing.assert_allclose(t.r, np.linspace(1.0,2.0,3))
+
+        #test creation with all params and edges=False
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=2.5, fmax=1.75, fcut=2.0, edges=False)
+        self.assertEqual(t._nbins, 4)
+        self.assertAlmostEqual(t._rmin, 1.0)
+        self.assertAlmostEqual(t._rmax, 2.5)
+        self.assertAlmostEqual(t.fmax, 1.75)
+        self.assertAlmostEqual(t.fcut, 2.0)
+        self.assertAlmostEqual(t.dr, 0.375)
+        np.testing.assert_allclose(t.r, 1.0+0.375*(np.arange(4)+0.5))
+
+        #test creation with invalid nbins/rmin/rmax setup
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2.5, rmin=0.5, rmax=1.5)
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2, rmin=1.0, rmax=0.5)
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2, rmin=-1.0, rmax=1.0)
+
+    def test_potential(self):
+        """Test energy and force methods"""
+        p1 = QuadPot(types=('1',), params=('m',), default={'m':2})
+        p2 = QuadPot(types=('1','2'), params=('m',), default={'m':1})
+        p_all = [p1, p2]
+
+        #test energy method
+        #test with edges=True
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0)
+        u = t.energy(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([12,3,0,3,12]))
+        u = t.energy(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([4,1,0,1,4]))
+        #test with edges=False
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0, edges=False)
+        u = t.energy(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([6.75,0.75,0.75,6.75]))
+        u = t.energy(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([2.25,0.25,0.25,2.25]))
+
+        #test force method
+        #test with edges=True
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0)
+        f = t.force(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([12,6,0,-6,-12]))
+        f = t.force(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([4,2,0,-2,-4]))
+        #test with edges=False
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0, edges=False)
+        f = t.force(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([9,3,-3,-9]))
+        f = t.force(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([3,1,-1,-3]))
+
+    def test_regularize(self):
+        """Test regularize_force method cases"""
+        p1 = QuadPot(types=('1',), params=('m',), default={'m':2})
+        p2 = QuadPot(types=('1','2'), params=('m',), default={'m':1})
+        p_all = [p1, p2]
+
+        #test without fmax or fcut
+        t = potential.Tabulator(nbins=4, rmin=0.5, rmax=2.5)
```

Comment:
Similarly down here, can you try making just one tabulator and then setting its properties directly?